### PR TITLE
[008] - Homework 8 - Streams and Lambdas

### PR DESCRIPTION
**Pull request by yalenna**

### CODE REVIEW

**Review by tintinfairy — CHANGES_REQUESTED** [review]

File: src/homework8/Task1.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Task1 {
+    public static void main(String[] args) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter("./src/Homework7/Resources/output.txt"))) {
```
[the remark this answers is not in the document]
Comment (yalenna) [reply src/homework8/Task1.java]: I replaced resources package to `src` directory. Also I'll update paths in previous HW7. Also updated paths in Tasks 1, 2, 4, 5.

File: src/homework8/Task8.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package homework8;
+
+interface SumFunction {
```
[the remark this answers is not in the document]
Comment (yalenna) [reply src/homework8/Task8.java]: Added.

File: src/homework8/Task9.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package homework8;
+
+@FunctionalInterface
+interface NumberDivisibleBy31 {
```
[the remark this answers is not in the document]
Comment (yalenna) [reply src/homework8/Task9.java]: Done.

File: src/homework8/Task7.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package homework8;
```
[the remark this answers is not in the document]
Comment (yalenna) [reply src/homework8/Task7.java]: Done.

File: src/homework8/Task4.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Task4 {
+    public static void main(String[] args) throws IOException {
```
Comment (tintinfairy) [inline src/homework8/Task4.java:11]: FYI `throws` is not necessary here, cause you have a catch block.

**Comment by tintinfairy** [comment]
Well done!
This PR can be merged.

**Review by tintinfairy — APPROVED** [review]

**Pull request merged**